### PR TITLE
Insights for a Specific Course

### LINKED ISSUE #89: Create "getCourseReport" function for querying and displaying relevant course statistics

### PR DESCRIPTION
**Pull request by oompas**
Function not 100% done/finalized, don't merge yet <3

### CODE REVIEW

File: functions/src/callable/reports.ts [hunk 1]
```diff
@@ -9,6 +9,46 @@ import {
     getCollectionDocs, QuizAttemptDocument
 } from "../helpers/database";
 
+interface CourseAttempt {
+    courseId: string;
+    endTime: Date;
+    pass: boolean | null;
+    startTime: Date;
```
Comment (reid-moffat) [inline functions/src/callable/reports.ts]: Note: Whenever working with Firestore, it has it's own Timestamp object (firestore.Timestamp, https://firebase.google.com/docs/reference/node/firebase.firestore.Timestamp)

File: functions/src/callable/reports.ts [hunk 2]
```diff
@@ -9,6 +9,46 @@ import {
     getCollectionDocs, QuizAttemptDocument
 } from "../helpers/database";
 
+interface CourseAttempt {
```
Comment (reid-moffat) [inline functions/src/callable/reports.ts]: See functions/rc/helpers/database.ts, these interfaces are already defined

File: functions/src/callable/reports.ts [hunk 3]
```diff
@@ -92,39 +134,50 @@ const getCourseReports = onCall(async (request) => {
             throw new HttpsError('internal', "Error getting course reports, please try again later");
         });
 
-    const enrollments = await getCollectionDocs(DatabaseCollections.EnrolledCourse) as EnrolledCourseDocument[];
-    const courseAttempts = await getCollectionDocs(DatabaseCollections.CourseAttempt) as CourseAttemptDocument[];
-    const quizAttempts = await getCollectionDocs(DatabaseCollections.QuizAttempt) as QuizAttemptDocument[];
+    const enrollments = await getCollection(DatabaseCollections.EnrolledCourse)
```
Comment (reid-moffat) [inline functions/src/callable/reports.ts]: Why was this reverted?

File: functions/src/callable/reports.ts [hunk 4]
```diff
@@ -137,4 +190,164 @@ const getCourseReports = onCall(async (request) => {
     }).sort((a, b) => b.numEnrolled - a.numEnrolled);
 });
 
-export { getUserReports, getCourseReports };
+
+/**
```
Comment (reid-moffat) [inline functions/src/callable/reports.ts:144]: This part is good, can keep it, but the rest changes above aren't needed

**Review by reid-moffat — CHANGES_REQUESTED** [review]

**Comment by oompas** [comment]
Function fully completed and changes made to fix the errors from the git merge & revert :)

**Review by reid-moffat — APPROVED** [review]

**Pull request merged**